src/Marten.Testing/StoreOptionsTests.cs

```diff
@@ -102,6 +102,18 @@ public void default_ddl_rules()
             options.DdlRules.UpsertRights.ShouldBe(SecurityRights.Invoker);
         }
 
+        [Fact]
+        public void ensure_patch_system_transform_functions_and_feature_schemas_are_added_only_once()
+        {
+            var options = new StoreOptions();
+            options.Connection(ConnectionSource.ConnectionString);
+
+            var store1 = new DocumentStore(options);
+            // pass with the same options and check it does not throw ArgumentException
+            // "An item with the same key has already been added. Key: <transform function name/feature schema name>"
+            var store2 = new DocumentStore(options);
```

Comment:
@mysticmind could you add explicit check with `Should.NotThrow` method? Then the comment wouldn't be needed.

Reply:
@oskardudycz god suggestion, I have added `Should.NotThrow`. Albeit, it does not have the means to pass a specific exception.